app/services/AuthService.ts

```diff
@@ -67,4 +72,99 @@ export const AuthService = {
 
     return payload;
   },
+
+  /**
+   *
+   *
+   * @param {String} token Authorization token attached to the HTTP header
+   * @return {users} Decoded jsonwebtoken
+   * @throws {JwtMalformedException} Throws error when token is malformed or empty
+   */
+
+  //will Sign and validate JWT token
+  async jwtSign(uid: string): Promise<string> {
+    const jwtAll = jwt.sign(
+      {
+        //will sign token for up to 24 hours
+        exp: Math.floor(Date.now() / 1000) + 24 * 60 * 60,
+        uid: uid,
+      },
+      config.app.secret
+    );
+
+    const found = await prisma.users.findFirst({
+      where: {
+        uid: uid,
+      },
+    });
+    if (found === null) {
+      // will create JWT token if user id is not found
+      const check = await prisma.users.create({
+        data: {
+          uid: uid,
+          token: jwtAll,
+        },
+      });
+    } else {
+      // will update JWT token if user id is found
+      const check = await prisma.users.update({
+        where: {
+          uid: uid,
+        },
+        data: {
+          token: jwtAll,
+        },
+      });
+    }
+
+    return jwtAll;
+  },
+
+  //Implement ldap system for login to the salisbury university ldap
+  async ldapJs(uid: string, pass: string): Promise<string> {
+    //creates connection with henson LDAPJS system
+    const client = ldapjs.createClient({
+      url: config.app.ldap,
+    });
+
+    client.on("error", (err) => {
+      //console.log("test")
+      // handle connection error
+    });
+
+    var user = "";
+
+    await client.bind(
+      "uid=" + uid + ",ou=users,dc=hslinux,dc=Salisbury,dc=edu",
```

Comment:
Not sure about this one, but maybe putting the dn to the .env might be better way